src/riak_cs_xml.erl

```diff
@@ -143,26 +150,34 @@ owner_content({OwnerName, OwnerId}) ->
     [make_external_node('ID', OwnerId),
      make_external_node('DisplayName', OwnerName)].
 
-list_objects_response_to_xml(Resp) ->
-    KeyContents = [key_content_to_xml(Content) ||
-                   Content <- (Resp?LORESP.contents)],
-    CommonPrefixes = [common_prefix_to_xml(Prefix) ||
-                         Prefix <- Resp?LORESP.common_prefixes],
-    Contents = [make_external_node('Name', Resp?LORESP.name),
-                make_external_node('Prefix', Resp?LORESP.prefix),
-                make_external_node('Marker', Resp?LORESP.marker)] ++
+list_objects_response_to_simple_form(Resp) ->
+    KeyContents = [{'Contents', key_content_to_simple_form(Content)} ||
+                      Content <- (Resp?LORESP.contents)],
+    CommonPrefixes = [{'CommonPrefixes', [{'Prefix', [CommonPrefix]}]} ||
+                         CommonPrefix <- Resp?LORESP.common_prefixes],
+    Contents = [{'Name',       [Resp?LORESP.name]},
+                {'Prefix',     [Resp?LORESP.prefix]},
+                {'Marker',     [Resp?LORESP.marker]}] ++
                 %% use a list-comprehension trick to only include
                 %% the `NextMarker' element if it's not `undefined'
-               [make_external_node('NextMarker', NextMarker) ||
-                NextMarker <- [Resp?LORESP.next_marker],
-                NextMarker =/= undefined,
-                Resp?LORESP.is_truncated] ++
-               [make_external_node('MaxKeys', Resp?LORESP.max_keys),
-                make_external_node('Delimiter', Resp?LORESP.delimiter),
-                make_external_node('IsTruncated', Resp?LORESP.is_truncated)] ++
```

Comment:
Sorry for stepping in... this line may be necessary, it was added for bug fix at https://github.com/basho/riak_cs/commit/45387fd2afb9be80379caf25eacb7cc4eebf026b?w=1